### PR TITLE
Print callstacks for errors in inlined functions

### PR DESCRIPTION
**Pull request by mppf**
The goal of this PR is to improve the error message for
`test/domains/marybeth/test_compare_range.chpl` as a follow-up to PR
#16823.

This PR adjusts the compiler error logic to print the `In function`
header and callstack in more cases. Previously, we weren't printing it
for inline functions but I can't rationalize that. There is already a
workaround for initializers for this. The behavior started in
5c147eb98e02948ec29a0b078edb484561dcca14 but hiding the function doesn't
make sense to me - the error still occurs within the inline function -
and the user might have to fix that function.

Reviewed by @e-kayrakli - thanks!

- [x] full local testing

### CODE REVIEW

File: compiler/util/astlocs.cpp [hunk 1]
```diff
@@ -70,6 +70,10 @@ astlocMarker::~astlocMarker() {
 //   not in a function in beginning with chpl__
```
Comment (cassella) [inline compiler/util/astlocs.cpp:70]: While you're here, should this be `not in a function beginning with ...`?

Comment (mppf) [reply compiler/util/astlocs.cpp:70]: I'm having trouble understanding what you are asking? It seems to already say that.

Comment (cassella) [reply compiler/util/astlocs.cpp:70]: There's an extra `in` in there.

**Comment by e-kayrakli** [comment]
Why are we adding `--no-print-callstack-on-error` in bunch of tests?

File: compiler/util/astlocs.cpp (hunk 1 above)
Comment (mppf) [reply compiler/util/astlocs.cpp]: Gotcha, thanks. Fixed.

**Comment by mppf** [comment]
> Why are we adding --no-print-callstack-on-error in bunch of tests?

It seemed like an easier way to update some of the tests. For tests with a deep call stack in the standard modules, we'll have to add a prediff to remove most of the callstack (or update the test when line numbers / non-public function names change) and it seemed easier/better to just leave out the call stack in those cases. 

IIRC the tests where I did this were with errors coming from `_write_one_internal` which has quite a big call stack.

**Comment by e-kayrakli** [comment]
I see. That makes sense, thanks!

**Review by e-kayrakli — APPROVED** [review]

**Pull request merged**